src/main/java/frc/robot/subsystems/hood/HoodIOReal.java

```diff
@@ -0,0 +1,49 @@
+package frc.robot.subsystems.hood;
+
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.MotionMagicExpoTorqueCurrentFOC;
+import com.ctre.phoenix6.hardware.TalonFX;
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+
+public class HoodIOReal implements HoodIO {
+    private final TalonFX motor;
+    private final DutyCycleEncoder absoluteEncoder = new DutyCycleEncoder(9); // TODO: to be changed
+    private final MotionMagicExpoTorqueCurrentFOC positionControl =
+            new MotionMagicExpoTorqueCurrentFOC(0);
+    private final DutyCycleOut dutyCycleOut = new DutyCycleOut(0);
+
+    public HoodIOReal() {
+        motor = new TalonFX(6); // TODO: to be changed
+        motor.getConfigurator().apply(HoodConstants.motorConfiguration);
+    }
+
+    @Override
+    public void updateInternalEncoder() {
```

Comment:
Just noticed a problem with this, the higher-level command requires the subsystem, which means we won't be able to run other commands while resetting the encoder (which I believe will happen constantly).

Reply:
If we set this to a default command will it fix it?

Reply:
It may be a default command, but you still can't run two commands in parallel. Notice also that default commands must require a subsystem.

Reply:
bcd88af
The command will run with trigger that is always true in RobotContainer

Reply:
Great solution! Note you can use the slightly shorter Commands.run

Reply:
Wouldn't it require the subsystem?